### PR TITLE
Change how ruby proxy is invoked.

### PR DESCRIPTION
**Pull request by nanliu**

### CODE REVIEW

File: lib/puppet_x/bodeco/util.rb [hunk 1]
```diff
@@ -48,7 +53,7 @@ def follow_redirect(uri, option = { limit: FOLLOW_LIMIT }, &block)
                     else
                       { use_ssl: false }
                     end
-        Net::HTTP.start(uri.host, uri.port, http_opts) do |http|
+        Net::HTTP.start(uri.host, uri.port, http_opts, @proxy_addr, @proxy_port) do |http|
```
Comment (TeunRoefs) [inline lib/puppet_x/bodeco/util.rb]: @nanliu Almost there, changing Net::HTTP.start(uri.host, uri.port, http_opts, @proxy_addr, @proxy_port) to Net::HTTP.start(uri.host, uri.port, @proxy_addr, @proxy_port, http_opts) will fix the issue on Windows.

**Comment by TeunRoefs** [comment]
@nanliu Ready for merge I guess? Is there anything I need to do?

**Comment by TeunRoefs** [comment]
Any update on this PR, kinda need this functionality and do not want to fork the module.

**Comment by alexjfisher** [comment]
LGTM
```
alex@phantom:~$ cat test.pp
archive { '/tmp/afisher.test':
  ensure       => present,
  source       => 'https://github.com/voxpupuli/puppet-archive/archive/master.zip',
  cleanup      => false,
  extract      => false,
  proxy_server => 'http://localhost:3128',
  provider     => 'ruby',
}

alex@phantom:~$ sudo tail /var/log/squid/access.log
alex@phantom:~$ rm /tmp/afisher.test ;puppet apply -t test.pp; sudo tail /var/log/squid/access.log
Info: Loading facts
Info: Loading facts
Notice: Compiled catalog for phantom in environment production in 0.13 seconds
Info: Applying configuration version '1495047363'
Notice: /Stage[main]/Main/Archive[/tmp/afisher.test]/ensure: download archive from https://github.com/voxpupuli/puppet-archive/archive/master.zip to /tmp/afisher.test  without cleanup
Notice: Applied catalog in 1.22 seconds
1495047364.754    749 127.0.0.1 TCP_TUNNEL/200 232053 CONNECT codeload.github.com:443 - HIER_DIRECT/192.30.253.120 -
1495047364.754   1161 127.0.0.1 TCP_TUNNEL/200 6126 CONNECT github.com:443 - HIER_DIRECT/192.30.253.113 -
```

File: lib/puppet_x/bodeco/util.rb [hunk 2]
```diff
@@ -25,10 +25,15 @@ def initialize(_url, options)
...
         @cookie = options[:cookie]
         @insecure = options[:insecure]
-        proxy_server = options[:proxy_server]
-        proxy_type = options[:proxy_type]
 
-        ENV["#{proxy_type}_proxy"] = proxy_server
+        if options[:proxy_server]
+          uri = URI(options[:proxy_server])
+          unless uri.scheme
+            uri = URI("#{options[:proxy_type]}://#{options[:proxy_server]}")
```
Comment (alexjfisher) [inline lib/puppet_x/bodeco/util.rb]: This line should never be hit?

I still think we've got confusion about what `proxy_type` is supposed to do.  At the moment, it behaves differently depending on the provider.  Then according to the README and original proxy PR, `proxy_server` is supposed to be of the form `http://my_proxy.example.com:3128`  I think it's confusing to have some providers treat the parameters differently.

**Comment by krjensen** [comment]
Could we get this merged and created a new release? - need this functionality

**Comment by edestecd** [comment]
I can confirm this works as well on windows...

**Pull request merged**